components/core/src/clp/streaming_compression/lzma/Compressor.cpp

```diff
@@ -0,0 +1,214 @@
+#include "Compressor.hpp"
+
+#include <array>
+#include <cstddef>
+#include <cstdint>
+#include <cstring>
+
+#include <lzma.h>
+#include <spdlog/spdlog.h>
+
+#include "../../ErrorCode.hpp"
+#include "../../FileWriter.hpp"
+#include "../../TraceableException.hpp"
+#include "../../type_utils.hpp"
+#include "Constants.hpp"
+
+namespace clp::streaming_compression::lzma {
+using clp::size_checked_pointer_cast;
+
+auto Compressor::init_lzma_encoder(LzmaStream* strm, int compression_level, size_t dict_size)
```

Comment:
I might consider turn this into a local function in an anonymized namespace

Reply:
The thing is it uses exceptions and `OperationFailed` that are defined within the compressor class.